### DISCUSSION TITLE
Github Desktop and VS Code Insiders

### DISCUSSION THREAD

**Discussion by whippingdot** [community/community#22238] [New to GitHub]
<p>In github desktop, can you select VS Code Insider’s addition as your editor cause I couldn’t find it.</p>

**Comment by billythekid** [comment]
<p>Hi <a class="mention" href="/whippingdot">@whippingdot</a>,</p>
<p>The supported editors for Desktop can be found here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/desktop/installing-and-configuring-github-desktop/configuring-a-default-editor" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/desktop/installing-and-configuring-github-desktop/configuring-a-default-editor" target="_blank" rel="noopener nofollow ugc">Configuring a default editor - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I’ve seen a couple of old issues about VS Code Insiders in the desktop repository that suggest it’s supported:</p>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/desktop/desktop/issues/5281" target="_blank" rel="noopener nofollow ugc">github.com/desktop/desktop</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/desktop/desktop/issues/5281" target="_blank" rel="noopener nofollow ugc">add support for detecting new User Install of VS Code Insiders</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2018-07-30" data-time="12:53:26" data-timezone="UTC">12:53PM - 30 Jul 18 UTC</span>
      </div>

        <div class="date">
          closed <span class="discourse-local-date" data-format="ll" data-date="2018-08-18" data-time="11:29:51" data-timezone="UTC">11:29AM - 18 Aug 18 UTC</span>
        </div>

      <div class="user">
        <a href="https://github.com/CharlesStover" target="_blank" rel="noopener nofollow ugc">
          <img alt="CharlesStover" src="https://user-images.githubusercontent.com/330170/181069680-b276e3e3-2628-45be-a023-e82ef3d84a65.jpeg" class="onebox-avatar-inline" width="20" height="20">
          CharlesStover
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">File -&gt; Options -&gt; Advanced -&gt; External editor
Describe the solution you'd like
I believe VS Code Insiders is supported, however Microsoft has...</p>
</div>

<div class="labels">
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">help wanted</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">integrations</span>
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">windows</span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<br>
<aside class="onebox githubpullrequest">
  <header class="source">
      <a href="https://github.com/desktop/desktop/pull/3441" target="_blank" rel="noopener nofollow ugc">github.com/desktop/desktop</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/desktop/desktop/pull/3441" target="_blank" rel="noopener nofollow ugc">support VSCode Insiders build alongside Stable</a>
    </h4>

    <div class="branches">
      <code>desktop:master</code> ← <code>say25:vs-code-insiders-side-by-side</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2017-11-28" data-time="07:31:09" data-timezone="UTC">07:31AM - 28 Nov 17 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/say25" target="_blank" rel="noopener nofollow ugc">
          <img alt="say25" src="https://user-images.githubusercontent.com/330170/181069681-f60cc26d-2239-4e73-acb8-356196401601.jpeg" class="onebox-avatar-inline" width="20" height="20">
          say25
        </a>
      </div>

      <div class="lines" title="9 commits changed 3 files with 83 additions and 17 deletions">
        <a href="https://github.com/desktop/desktop/pull/3441/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+83</span>
          <span class="removed">-17</span>
        </a>
      </div>
    </div>

  </div>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p><s>I’m asking the team involved with Desktop for more information and I’ll get back to you as soon as I hear back on this.</s></p>
<p>The way this is detected differs between Windows or Mac installs.</p>
<hr>
<p><strong>If you’re using Windows</strong></p>
<p>I’ve had it confirmed that this <a href="https://github.com/desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/win32.ts#L21" rel="noopener nofollow ugc">should be available in the selection menu</a> if it’s installed in the default way using the windows installer, but may not appear if it’s been installed by way of a portable install or using chocolatey or similar.</p>
<p>These are the registry locations it uses to detect this, if you’re confident with registry entries you can check if these exist:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/win32.ts#L101-L125" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/win32.ts#L101-L125" target="_blank" rel="noopener nofollow ugc">desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/win32.ts#L101-L125</a></h4>
<pre class="onebox"><code class="lang-ts"><ol class="start lines" start="101" style="counter-reset: li-counter 100 ;">
<li>return [</li>
<li>  // 64-bit version of VSCode (user) - provided by default in 64-bit Windows</li>
<li>  {</li>
<li>    key: HKEY.HKEY_CURRENT_USER,</li>
<li>    subKey:</li>
<li>      'SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\{217B4C08-948D-4276-BFBB-BEE930AE5A2C}_is1',</li>
<li>  },</li>
<li>  // 32-bit version of VSCode (user)</li>
<li>  {</li>
<li>    key: HKEY.HKEY_CURRENT_USER,</li>
<li>    subKey:</li>
<li>      'SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\{26F4A15E-E392-4887-8C09-7BC55712FD5B}_is1',</li>
<li>  },</li>
<li>  // 64-bit version of VSCode (system) - was default before user scope installation</li>
<li>  {</li>
<li>    key: HKEY.HKEY_LOCAL_MACHINE,</li>
<li>    subKey:</li>
<li>      'SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\{1287CAD5-7C8D-410D-88B9-0D1EE4A83FF2}_is1',</li>
<li>  },</li>
<li>  // 32-bit version of VSCode (system)</li>
</ol></code></pre>

  This file has been truncated. <a href="https://github.com/desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/win32.ts#L101-L125" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<hr>
<p><strong>If this is a Mac install</strong></p>
<p>We use the app bundle ID to detect its existence on the system:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/darwin.ts#L46" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/darwin.ts#L46" target="_blank" rel="noopener nofollow ugc">desktop/desktop/blob/a7ec771daff26a274c221c87e2a337765cfa2825/app/src/lib/editors/darwin.ts#L46</a></h4>
<pre class="onebox"><code class="lang-ts"><ol class="start lines" start="36" style="counter-reset: li-counter 35 ;">
<li>
<li>function getBundleIdentifiers(editor: ExternalEditor): ReadonlyArray&lt;string&gt; {</li>
<li>  switch (editor) {</li>
<li>    case ExternalEditor.Atom:</li>
<li>      return ['com.github.atom']</li>
<li>    case ExternalEditor.MacVim:</li>
<li>      return ['org.vim.MacVim']</li>
<li>    case ExternalEditor.VSCode:</li>
<li>      return ['com.microsoft.VSCode']</li>
<li>    case ExternalEditor.VSCodeInsiders:</li>
<li class="selected">      return ['com.microsoft.VSCodeInsiders']</li>
<li>    case ExternalEditor.VSCodium:</li>
<li>      return ['com.visualstudio.code.oss']</li>
<li>    case ExternalEditor.SublimeText:</li>
<li>      return ['com.sublimetext.3']</li>
<li>    case ExternalEditor.BBEdit:</li>
<li>      return ['com.barebones.bbedit']</li>
<li>    case ExternalEditor.PhpStorm:</li>
<li>      return ['com.jetbrains.PhpStorm']</li>
<li>    case ExternalEditor.PyCharm:</li>
<li>      return ['com.jetbrains.PyCharm']</li>
</ol></code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You can check if this is a resolvable path by running the following command in your terminal:</p>
<p><code>mdfind "kMDItemCFBundleIdentifier === 'com.microsoft.VSCodeInsiders'"</code></p>
<p>Kind regards,<br>
Billy</p>

**Comment by whippingdot** [comment]
<p>Oops, I checked my github Desktop now and it came. Thanks for the help. :smiley:. I am really sorry. I tried after some time earlier too but it didn’t come. It has just came after about 5 hrs. Thanks anyways.</p>

**Comment by billythekid** [comment]
<p>That’s great to hear, thanks for the update!</p>
